notebook/services/api/api.yaml

```diff
@@ -654,12 +639,18 @@ definitions:
       id:
         type: string
         format: uuid
-      notebook:
-        type: object
-        properties:
-          path:
-            type: string
-            description: path to notebook
+      path:
+        type: string
+        description: "The path to the session.  For console sessions, the path 
+         contains the cwd of the kernel and the name of the console session, 
+         which may be a uuid"
+      type:
+        type: string;
+        description: session type
+        enum:
+          - notebook
+          - file
+          - console
```

Comment:
Then maybe I'd leave it as `path`, and not `cwd`. For instance, with a notebook you ask for a kernel associated with a given notebook. There's no directory that's part of the request, nor should there be. Same goes for a file, I would think (we still don't have anything associating kernels with non-notebook files, but I'm guessing we might want it to be the same). But a console might be associated with a directory, as you point out.